extra_data/caldata.py

```diff
@@ -0,0 +1,1016 @@
+
+"""Interfaces to calibration constant data."""
+
+from datetime import datetime, date, time, timezone
+from enum import IntFlag
+from functools import lru_cache
+from os import getenv
+from pathlib import Path
+import re
+import socket
+
+import numpy as np
+import h5py
+
+from . import SourceNameError, PropertyNameError, MultiRunError, NoDataError
+
+from calibration_client import CalibrationClient
+from calibration_client.modules import Detector, PhysicalDetectorUnit, \
+    Parameter, Calibration, CalibrationConstantVersion
+
+
+__all__ = [
+    'BadPixels',
+    'CalCatError',
+    'AGIPD_CalibrationData',
+    'LPD_CalibrationData',
+    'DSSC_CalibrationData',
+    'JUNGFRAU_CalibrationData',
+    'PNCCD_CalibrationData',
+    'EPIX_CalibrationData',
+]
+
+
+class BadPixels(IntFlag):
+    """The European XFEL Bad Pixel Encoding."""
```

Comment:
Have we confirmed this is exactly the same for all detectors? It's the kind of thing I could imagine starting off the same and then growing different additional fields in different detectors.

Reply:
As far as I've understood, yes, this is indeed what was agreed upon for all detectors (@kakhahmed for confirmation). There are actually additional enums like `BadPixelsFF` for AGIPD-only calculations, which are then afterwards translated to this one. That is the reason I moved them recently to AGIPD-specific parts of `pycalibration`.

Some flags are only used for a single detector right now, but they appear broad enough to be sensible.

Reply:
yes, `BadPixels` is already used for all detectors

Reply:
Thanks!